src/main.rs

```diff
@@ -28,50 +29,73 @@ fn main() -> std::io::Result<()> {
 
     match result {
         Some(selection) => {
-            let selected_branch = options[selection];
-            let checkout_command = format!("git checkout {}", selected_branch);
-            spawn_command(&checkout_command);
+            let selected_branch = &options[selection];
+            checkout(repo, selected_branch)?;
         }
-        None => println!("Stayed on branch '{}'", current_branch),
+        None => match current_branch {
+            Some(branch) => println!("Stayed on branch '{}'", &branch),
+            None => println!("Doing nothing"),
+        },
     }
 
     Ok(())
 }
 
-/// Executes a command in the appropriate shell of the os, returning the output on stdout.
-fn exec_command(command: &str) -> String {
-    let output = if cfg!(target_os = "windows") {
-        Command::new("cmd")
-            .args(&["/C", command])
-            .output()
-            .expect("failed to execute process")
-    } else {
-        Command::new("sh")
-            .arg("-c")
-            .arg(command)
-            .output()
-            .expect("failed to execute process")
-    };
+fn checkout(repo: Repository, branch_name: &String) -> Result<()> {
+    let ref_name = format!("refs/heads/{}", branch_name);
+    repo.set_head(&ref_name)
+        .with_context(|| format!("Could not set head to {}", ref_name))?;
+    repo.checkout_head(Some(&mut CheckoutBuilder::default()))
+        .with_context(|| format!("Unable to check out branch {}", branch_name))?;
+    Ok(())
+}
+
+fn get_branch_options(sorted_branches: Vec<String>, current_branch: Option<&str>) -> Vec<String> {
+    let all_branches: Vec<String> = sorted_branches
+        .into_iter()
```

Comment:
To be honest, I don't know the difference between `.iter()` and `.into_iter()`.

Reply:
I assume `.into_iter()` does `.into()` as well as `.iter()`, which probably isn't needed here.